### PR TITLE
RC-2 Labrador parser

### PR DESCRIPTION
**Pull request by zetmann**

### CODE REVIEW

File: src/app/parser/BaseParser.ts [hunk 1]
```diff
@@ -75,7 +75,8 @@ abstract class BaseParser extends BaseElements implements Parser {
 		};
 
 		return Promise.resolve(Promise.props(workflow))
-		.then((a : ParserWorkflowPayload) : Article => ({
```
Comment (GrimStal) [inline src/app/parser/BaseParser.ts]: You can change it back. If we do not open "{}" in arrow function - that means we "return" selected result, so no need to use "return" param inside, just get off "{}"

Comment (GrimStal) [reply src/app/parser/BaseParser.ts]: PS - ES6

File: src/app/parser/impl/LabradorParser.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+		const error = 'Bad source response';
+		log(error);
+		return Promise.reject(new Error(error));
+	}
+
+	/**
+	 * Validating response
+	 */
+	validateApiResponse(rawArticle): boolean {
+		if (!rawArticle.result || ! Array.isArray(rawArticle.result) ) {
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: I am not sure, but in JS code convention we don't use a space after "!"

File: src/app/parser/impl/LabradorParser.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+			return false;
+		}
+		if (rawArticle.result.length !== 1 || !rawArticle.result[0]) {
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: Can be simplified. But I don't insist

File: src/app/parser/impl/LabradorParser.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+		}
+
+		return true;
+	}
+
+	/**
+	 * We still have html content that should be parsed as usual.
+	 */
+	getParsedContent() {
+		if (this.jsonArticle.bodytextHTML && this.jsonArticle.bodytextHTML) {
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: what is the difference?

File: src/app/parser/impl/LabradorParser.ts [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+					case 'h3':
+					case 'h4':
+					case 'h5':
+					case 'h6':
+						item = this.createSubHeadingEl($(elem).text());
+						break;
+					case 'a':
+						item = this.createLinkEl(elem.attribs.href, $(elem).text());
+						break;
+					case 'body':
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: Can be simplified to use break after all of them

File: src/app/parser/impl/LabradorParser.ts [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+					case 'head':
+						break;
+					case 'html':
+						break;
+					case 'p':
+					default:
+						item = this.createParagraphEl($(elem).text());
+						break;
+				}
+				if (item !== undefined) {
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: Maybe better to use (typeof item !== 'undefined')

File: src/app/parser/impl/LabradorParser.ts [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+					items.push(item);
+				}
+			});
+			return items;
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: Only return items in the end. If they are not set - it is empty array. No unusefull returns

File: src/app/parser/impl/LabradorParser.ts [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+		}
+
+		return [];
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: I mean here

File: src/app/parser/impl/LabradorParser.ts [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+	}
+
+	/**
+	 * Fetch data that shouldn't be parsed, but can be used directly from
+	 * JSON object.
+	 */
+	getProvidedContent() {
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: MB better

```
getProvidedContent(): ProvidedContent {
   return getContent(this.jsonArticle);
}
```

File: src/app/parser/impl/LabradorParser.ts [hunk 10]
```diff
@@ -0,0 +1,232 @@
+//
```
Comment (GrimStal) [inline src/app/parser/impl/LabradorParser.ts]: Check your files for this ))) It is LL project

File: src/app/parser/util/labradorparser/ProvidedContent.ts [hunk 11]
```diff
@@ -0,0 +1,205 @@
+//
```
Comment (GrimStal) [inline src/app/parser/util/labradorparser/ProvidedContent.ts]: Remove it

File: src/app/parser/util/labradorparser/ProvidedContent.ts [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program. If not, see <http://www.gnu.org/licenses/>.
+//
+import {pick} from 'lodash';
+//import ArticleItem from 'base/ArticleItem';
```
Comment (GrimStal) [inline src/app/parser/util/labradorparser/ProvidedContent.ts]: remove unused imports

File: src/app/parser/util/labradorparser/ProvidedContent.ts [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+		name: raw.firstname + ' ' + raw.lastname,
+	};
+}
+
+/**
+ * Title getter.
+ */
+function getTitle (rawData) : TitleInterface {
+	return {
+		text: rawData.title ? rawData.title : '',
```
Comment (GrimStal) [inline src/app/parser/util/labradorparser/ProvidedContent.ts]: text: rawData.title || '',

File: src/app/parser/util/labradorparser/ProvidedContent.ts [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+		type: ArticleItemType.MainTitle,
+	};
+}
+
+/**
+ * Subtitle getter.
+ */
+function getSubtitle (rawData) : SubTitleInterface {
+	return {
+		text: rawData.subtitle ? rawData.subtitle : '',
```
Comment (GrimStal) [inline src/app/parser/util/labradorparser/ProvidedContent.ts]: text: rawData.subtitle || '',

File: src/app/parser/util/labradorparser/ProvidedContent.ts [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+		type: ArticleItemType.SubTitle,
+	};
+}
+
+/**
+ * Bunch image getter.
+ */
+function getImages (rawArticle) : ImageInterface[] {
+	const defaultHeight = 400;
+	const featuredID = getFeaturedID(rawArticle);
```
Comment (GrimStal) [inline src/app/parser/util/labradorparser/ProvidedContent.ts]:
```
const {image} = rawArticle.contentBoxes;
return image.map(img => {
...
```

**Comment by phgroe** [comment]
Base architecture for Labrador parser prepared in #551 - implementation probably following

**Pull request closed without merging**